src/index.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as handler from './github-handler';
+import {
+  Changes,
+  Description,
+  Logger,
+  Octokit,
+  GitHubPrUserOptions,
+  RepoDomain,
+  BranchDomain,
+} from './types';
+import {logger, setupLogger} from './logger';
+import {addPrOptionDefaults} from './default-options-handler';
+
+/**
+ * Make GitHub Pull Request with a set of changes applied on top of primary branch HEAD.
+ * Rethrows an HttpError if Octokit GitHub API returns an error. HttpError Octokit access_token and client_secret headers redact all sensitive information.
+ * @param {Octokit} octokit The authenticated octokit instance.
+ * @param {Changes} changes A set of changes.
+ * @param {GitHubPrUserOptions} gitHubOptions The configuration for interacting with GitHub provided by the user.
+ * @param {Logger} logger The logger instance (optional).
+ * @returns {Promise<void>}
+ */
+async function makePr(
```

Comment:
Can we make all the `Pr` variables/classes into `PullRequest`? Generally, we should avoid abbreviations in variable/method names.

Additionally, does the update/rebase an existing pull request? If so, it's not clear from the docs or the method name.